Spigot-API-Patches/0127-EntityTransformedEvent.patch

```diff
@@ -0,0 +1,66 @@
+From 6697a68f368d1f850dfda38f71033d0c93e60769 Mon Sep 17 00:00:00 2001
+From: Anthony MacAllister <anthonymmacallister@gmail.com>
+Date: Thu, 26 Jul 2018 15:28:53 -0400
+Subject: [PATCH] EntityTransformedEvent
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/event/entity/EntityTransformedEvent.java b/src/main/java/com/destroystokyo/paper/event/entity/EntityTransformedEvent.java
+new file mode 100644
+index 0000000..28df627
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/event/entity/EntityTransformedEvent.java
+@@ -0,0 +1,51 @@
++package com.destroystokyo.paper.event.entity;
++
++
++import org.bukkit.entity.Entity;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++
++/**
++ * Fired when an entity transforms into another entity
++ * <p>
++ * If the event is cancelled, the entity will not transform
++ */
++public class EntityTransformedEvent extends EntityEvent implements Cancellable {
++    private static final HandlerList handlers = new HandlerList();
++    private boolean cancelled;
++    private Entity transformed;
++
++    public EntityTransformedEvent(Entity entity, Entity transformed) {
```

Comment:
As suggested, we need a TransformReason enum. Put the enum inside this class like other events do.

I'm ok with starting and merging just Drowned to begin with, but then we can add the other existing transformations too.

Reply:
We're working on it ^_^

Reply:
Sorry for the delay of this. Took longer than I wanted and it is still a learning curve with nms with me. I will get there eventually. @BillyGalbreath helped out a lot =D